### DISCUSSION TITLE
Create a CA without a user-in-the-loop

### DISCUSSION THREAD

**Discussion by mkgvt** [smallstep/certificates#837] [Q&A] [answered]
How do I automate the creation of a SmallStep CA without needing a user to explicitly answer the questions asked by `step ca init`?

Specifically: I am setting up a Terraform configuration to spin up a CA without the need for human interaction. I can see that there are two variations in this context:

1. Setting up the CA for the first time, which prompts the question above.
2. Spinning up the CA again with the same configuration as before. My assumption here is that the state would have to be preserved between activations.

-M

**Comment by mkgvt** [comment, marked as answer]
Found the answer. Options to `step ca init` can be specified via command line options.

**Reply by tashian** [reply]
Yes and if you need to get into things that can't be controlled via the command line options (eg. commonly  default validity periods are something people want to adjust), you can always tweak the `config.json` with `jq`. Not ideal, I know, but doable.
